### PR TITLE
Cache response from yt_video_metadata view

### PR DESCRIPTION
**Pull request by magajh**
This PR allows caching the response from the youtube video metadata view.

### CODE REVIEW

**Review by mariajgrimaldi — APPROVED** [review]

**Review by felipemontoya — COMMENTED** [review]
I was looking at the cache_page function at:
https://docs.djangoproject.com/en/3.1/topics/cache/

But it is not clear for me that the different IDs of different videos will be cached separately.

Are you sure it works in this scenario?

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -298,40 +298,50 @@ def load_metadata_from_youtube(video_id, request):
...
-                        metadata = res_json
-                    else:
-                        logging.warning(u'Unable to find the items in response. Following response '
+    cache_key = "yt_video_metadata_cache.{video}".format(video=video_id)
+    metadata = cache.get(cache_key, {})
+    status_code = 200
+
+    if not metadata:  # pylint: disable=too-many-nested-blocks
+        status_code = 500
+        if video_id and settings.YOUTUBE_API_KEY and settings.YOUTUBE_API_KEY != 'PUT_YOUR_API_KEY_HERE':
```
Comment (mariajgrimaldi) [inline lms/djangoapps/courseware/views/views.py:307]: Why not use getattr?

File: lms/djangoapps/courseware/views/views.py [hunk 2]
```diff
@@ -298,40 +298,50 @@ def load_metadata_from_youtube(video_id, request):
...
-                    res_json = res.json()
-                    if res_json.get('items', []):
-                        metadata = res_json
-                    else:
-                        logging.warning(u'Unable to find the items in response. Following response '
+    cache_key = "yt_video_metadata_cache.{video}".format(video=video_id)
+    metadata = cache.get(cache_key, {})
+    status_code = 200
+
+    if not metadata:  # pylint: disable=too-many-nested-blocks
```
Comment (mariajgrimaldi) [inline lms/djangoapps/courseware/views/views.py:305]: maybe you could remove `disable=too-many-nested-blocks` by adding 
if metadata return metadata, status_code
... rest of code
return metadata, status_code

???

Comment (magajh) [reply lms/djangoapps/courseware/views/views.py]: I changed this. Thank you

**Review by felipemontoya — APPROVED** [review]

**Pull request merged**